### ISSUE TITLE
bd CPS: flow example broken from roughly 2004 to 2007

### ISSUE THREAD

**Issue by bdecon** [bdecon/econ_data#58]
The flow example does not work for the years 1995 to 1997--this probably comes from adjustments made to the UID variable, as it had previously worked. Also, the related years data dictionary contains some odd characters, which could be affecting the results.

**Comment by bdecon** [comment]
Multiple issues here. One is solved: the HRHHID2 variable in may04dd.txt contains parentheses, which are not read by regex because they needed to be escaped. That is, regex needed "HRHHID \(partII\)". 

Still, some months aren't working correctly.

**Comment by bdecon** [comment]
Major remaining issue is the one year period from May 2004 to April 2005. It is because the HRHHID2 variable defined in the May 2004 forward data does not match with the previous HRHHID2 calculated manually pre May 2004. It is frustrated because all of this worked in a previous version of the bd CPS and I'm not sure why it worked then. 

The data dictionary suggests that I use only HRHHID to match backwards for these, but that doesn't seem to be what I had done in the previous version of the bd CPS that worked. I think the problem is coming from the way I define HRHHID2, because that is something that changed between the previous and current version. 

In the old version, I copied the function to create HRHHID2 directly from Tom Augspurger's pycps, but in the new version I wrote my own function that was basically a refactored version of his. I tried to get the old function to work with the new bd CPS, but with no luck yet.
